src/main.ts

```diff
@@ -53,35 +53,40 @@ async function run(): Promise<void> {
     await installTool();
 
     const projectDir: string = argv.projectDir ?? core.getInput("projectDir");
-    const projectsFilter: string = argv.projectsFilter ?? core.getInput("projectsFilter");
+    const solutionFileName: string = argv.solutionFileName ?? core.getInput("solutionFileName");
+
+    const solutionPath = path.join(projectDir, solutionFileName);
+
+    const excludeProjects: string = argv.excludeProjects ?? core.getInput("excludeProjects");
     const allowedLicenses: string = argv.allowedLicenses ?? core.getInput("allowedLicenses");
     const exportDir = argv.exportDir ?? core.getInput("exportDir");
     const tempExportDir = path.join(os.tmpdir(), uuidv4());
 
-    const args = ["--input", projectDir, "--unique"];
-    await addProjectsFilter(args, projectsFilter);
+    const args = ["--input", solutionPath];
+
+    await addExcludeProjects(args, excludeProjects);
     await addAllowedLicenses(args, allowedLicenses);
     await addLicenseUrlMappings(args, projectDir);
     await addLicensePackageMappings(args, projectDir);
+    await addIgnorePackages(args, projectDir);
     await addExportOptions(args, exportDir, tempExportDir);
-
-    await exec("dotnet-project-licenses", args);
+    await exec("nuget-license", args);
 
     await buildReport(tempExportDir, exportDir);
   } catch (error: unknown) {
     core.setFailed((error as { message: string }).message);
   }
 }
 
-async function addProjectsFilter(args: string[], projectsFilter: string) {
-  const projectsFilterFile = path.join(os.tmpdir(), "projectsFilter.json");
-  const projectsFilterList = chain(projectsFilter)
+async function addExcludeProjects(args: string[], excludeProjects: string) {
+  const excludeProjectsFile = path.join(os.tmpdir(), "excludeProjects.json");
+  const excludeProjectsList = chain(excludeProjects)
     .split(";")
     .filter((x) => x.length > 0)
     .value();
 
-  await fs.promises.writeFile(projectsFilterFile, JSON.stringify(projectsFilterList));
-  args.push("--projects-filter", projectsFilterFile);
+  await fs.promises.writeFile(excludeProjectsFile, JSON.stringify(excludeProjectsList));
```

Comment:
@datlam162 i think you dont need to write the excludeProjects to a file anymore as we pass it directly as parameter to the cli tool